src/Models/HydrostaticFreeSurfaceModels/mg_implicit_free_surface_solver.jl

```diff
@@ -20,7 +22,9 @@ The multigrid implicit free-surface solver.
 
 $(TYPEDFIELDS)
 """
-mutable struct MGImplicitFreeSurfaceSolver{S, V, F, R, C, D}
+mutable struct MGImplicitFreeSurfaceSolver{A, S, V, F, R, C, D}
+    "Architecture"
+    architecture :: A
     "The multigrid solver"
     multigrid_solver :: S
     "The vertically-integrated lateral areas"
```

Comment:
You don't need this right?

Reply:
Nice catch!

Reply:
Also @elise-palethorpe, for the fields below I'd suggest the somehow more explanatory docstrings:

```julia
    "The matrix constructors of the linear operator without the diagonal `Az / (g Δt²)` term"
    matrix_constructors :: C
    "The diagonal matrix term `Az / g`"
    diagonal :: D
```